src/backend_model.cc

```diff
@@ -431,6 +475,59 @@ TritonModel::SetConfiguredScheduler()
   return SetScheduler(std::move(scheduler));
 }
 
+Status
+TritonModel::SetBatchingStrategy(const std::string& batch_libpath)
+{
+  // Load library and functions.
+  std::unique_ptr<SharedLibrary> slib;
+  RETURN_IF_ERROR(SharedLibrary::Acquire(&slib));
+
+  RETURN_IF_ERROR(slib->OpenLibraryHandle(batch_libpath, &batch_dlhandle_));
+  RETURN_IF_ERROR(slib->GetEntrypoint(
+      batch_dlhandle_, "TRITONBACKEND_ModelBatchIncludeRequest",
+      true /* optional */, reinterpret_cast<void**>(&batch_incl_fn_)));
+  RETURN_IF_ERROR(slib->GetEntrypoint(
+      batch_dlhandle_, "TRITONBACKEND_ModelBatchInitialize",
+      true /* optional */, reinterpret_cast<void**>(&batch_init_fn_)));
+  RETURN_IF_ERROR(slib->GetEntrypoint(
+      batch_dlhandle_, "TRITONBACKEND_ModelBatchFinalize", true /* optional */,
+      reinterpret_cast<void**>(&batch_fini_fn_)));
+  RETURN_IF_ERROR(slib->GetEntrypoint(
+      batch_dlhandle_, "TRITONBACKEND_ModelBatcherFinalize",
+      true /* optional */, reinterpret_cast<void**>(&batcher_fini_fn_)));
+  TritonModelBatcherInitFn_t batcher_init_fn;
+  RETURN_IF_ERROR(slib->GetEntrypoint(
+      batch_dlhandle_, "TRITONBACKEND_ModelBatcherInitialize",
+      true /* optional */, reinterpret_cast<void**>(&batcher_init_fn)));
+
+  // If one custom batching function is defined, all must be.
+  if ((batch_incl_fn_ || batch_init_fn_ || batch_fini_fn_ || batcher_init_fn ||
+       batcher_fini_fn_) &&
+      !(batch_incl_fn_ && batch_init_fn_ && batch_fini_fn_ && batcher_init_fn &&
+        batcher_fini_fn_)) {
+    ClearHandles();
+    batcher_init_fn = nullptr;
```

Comment:
This is done in ClearHandles. This is just clearing the one that's a local variable (since we use call the init fn right after the checks pass, so we don't need to save it).

Reply:
I see, missed that it is a local variable, but can you make it member variable like the others for consistency?

Reply:
Sure, done.